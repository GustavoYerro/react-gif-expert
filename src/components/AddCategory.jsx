import { useState } from "react";
import PropTypes from "prop-types";

//rafc
export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('');

    // const onInputChange = ( event ) => {
    //     console.log( event.target.value );
    //     setinputValue( event.target.value );
    // }

    const onInputChange = ({target}) => {
        //console.log( target.value );
        setinputValue( target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();

        if(inputValue.trim().length <= 1) return;

        //setCategories(categories => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setinputValue('');
    }

    return (
        // <form onSubmit={ (event) => onSubmit(event)}>
        <form onSubmit={ onSubmit } aria-label="form">
            <input
            type="text"
            placeholder="Buscar gifs"
            value = { inputValue }
            onChange = { onInputChange }
        />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
