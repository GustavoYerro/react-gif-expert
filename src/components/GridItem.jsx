import PropTypes from "prop-types";

export const GridItem = ({title, url, id}) => {

    //console.log( props );
    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GridItem.protTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}