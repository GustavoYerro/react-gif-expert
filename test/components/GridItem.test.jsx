import { render, screen } from '@testing-library/react';
import { GridItem } from '../../src/components/GridItem';

describe('Pruebas en GridItem', () => { 

    const title = 'titulo';
    const url = 'https://url.com/soyunaurl.jpg';

    test('Debe hacer match con el snapshot', () => { 
    
        const { container } = render( <GridItem title={ title} url={ url } />);
        expect ( container ).toMatchSnapshot();

    });
    
    test('Debe mostrar el titulo', () => { 
         render( <GridItem title={ title} url={ url } />);
         expect( screen.getByText( title )).toBeTruthy();
    })

    test('debe mostrar la imagen con la url y el alt indicado', () => { 
        
        render( <GridItem title={ title} url={ url } />);
        //screen.debug();
        //expect( screen.getByRole('img').src ).toBe( url );
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })
})