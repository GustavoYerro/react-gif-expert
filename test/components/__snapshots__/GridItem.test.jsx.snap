// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GridItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://url.com/soyunaurl.jpg"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
